src/core/mesh/qgsmeshtracerenderer.cpp

```diff
@@ -647,37 +655,148 @@ bool QgsMeshStreamField::addPixelToChunkTrace( QPoint &pixel,
 void QgsMeshStreamlinesField::initField()
 {
   mField = QVector<bool>( mFieldSize.width() * mFieldSize.height(), false );
+  mDirectionField = QVector<char>( mFieldSize.width() * mFieldSize.height(), static_cast<char>( int( 0 ) ) );
   initImage();
 }
 
+void QgsMeshStreamlinesField::initImage()
+{
+  mTraceImage = QImage();
+  switch ( mVectorColoring.coloringMethod() )
+  {
+    case QgsInterpolatedLineColor::ColorRamp:
+    {
+      QSize imgSize = mFieldSize * mFieldResolution;
+      QgsRenderContext fieldContext = mRenderContext;
+
+      fieldContext.setMapToPixel( mMapToFieldPixel );
+      std::unique_ptr<QgsMeshLayerInterpolator> mScalarInterpolator(
+        new QgsMeshLayerInterpolator(
+          mTriangularMesh,
+          mMagValues,
+          mScalarActiveFaceFlagValues,
+          mDataType,
+          fieldContext,
+          imgSize ) );
+
+      QgsRasterShader *sh = new QgsRasterShader();
+      sh->setRasterShaderFunction( new QgsColorRampShader( mVectorColoring.colorRampShader() ) ); // takes ownership of fcn
+      QgsSingleBandPseudoColorRenderer renderer( mScalarInterpolator.get(), 0, sh );  // takes ownership of sh
+      if ( imgSize.isValid() )
+      {
+        std::unique_ptr<QgsRasterBlock> bl( renderer.block( 0, mOutputExtent, imgSize.width(), imgSize.height(), mFeedBack ) );
+        mTraceImage = bl->image();
+      }
+    }
+    break;
+    case QgsInterpolatedLineColor::SingleColor:
+    {
+      mTraceImage = QImage( mFieldSize * mFieldResolution, QImage::Format_ARGB32_Premultiplied );
+      QColor col = mVectorColoring.singleColor();
+      mTraceImage.fill( col );
+    }
+    break;
+  }
+
+  if ( !mTraceImage.isNull() )
+  {
+    mPainter.reset( new QPainter( &mTraceImage ) );
+    mPainter->setRenderHint( QPainter::Antialiasing, true );
+
+    mDrawingTraceImage = QImage( mTraceImage.size(), QImage::Format_ARGB32_Premultiplied );
+    mDrawingTraceImage.fill( Qt::transparent );
+    mDrawingTracePainter.reset( new QPainter( &mDrawingTraceImage ) );
+    mDrawingTracePainter->setRenderHint( QPainter::Antialiasing, true );
+  }
+}
+
+void QgsMeshStreamField::clearChunkTrace( std::list<QPair<QPoint, QgsMeshStreamField::FieldData> > &chunkTrace )
```

Comment:
Just curious, why do we use std::list in those functions instead of qt containers?